var/spack/repos/builtin/packages/trilinos-catalyst-ioss-adapter/package.py

```diff
@@ -17,7 +17,9 @@ class TrilinosCatalystIossAdapter(CMakePackage):
 
     depends_on("bison", type="build")
     depends_on("flex", type="build")
-    depends_on("paraview+mpi+python+osmesa")
+    depends_on("paraview+mpi+python")
+    depends_on("gl")
+    requires("^[virtuals=gl] osmesa", msg="OSMesa is required for paraview")
```

Comment:
We can't say:
```
depends_on("foo ^bar")
```
but we can say:
```
depends_on("foo")
depends_on("bar")
```
After all, the dependency on `bar` is enforced by this package in the example above, not by `foo`.

Reply:
I see, makes sense then. It is odd to have deps on things that it doesn't use. Maybe mark it as a `link,run` dep only to avoid adding it to the link line?

Reply:
See https://github.com/spack/spack/pull/44307/commits/40f6cd39095d14cebfe0fa165b2b757342f74563